snuba/utils/streams/kafka.py

```diff
@@ -194,14 +214,52 @@ def poll(self, timeout: Optional[float] = None) -> Optional[KafkaMessage]:
             else:
                 raise ConsumerError(str(error))
 
-        return KafkaMessage(
+        result = KafkaMessage(
             TopicPartition(message.topic(), message.partition()),
             message.offset(),
             message.value(),
         )
 
+        self.__offsets[result.stream] = result.offset + 1
+
+        return result
+
+    def tell(self) -> Mapping[TopicPartition, int]:
+        if self.__state in {KafkaConsumerState.CLOSED, KafkaConsumerState.ERROR}:
+            raise InvalidState(self.__state)
+
+        return self.__offsets
+
+    def __seek(self, offsets: Mapping[TopicPartition, int]) -> None:
+        if self.__state is KafkaConsumerState.ASSIGNING:
```

Comment:
Ok, that makes more sense — my goal here was to make the `seek` method the only API for moving these offsets around in callbacks, paving over the inconsistency that `seek` should be called in some circumstances, and `assign` should be called in others. The tests might be a clearer example of this, or some of the upcoming stuff that uses this will also hopefully make it a bit clearer.